### ISSUE TITLE
Hello, undefined (and no admin for you)

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/hpi#554]
Sometimes, the header doesn't get the user's name.  Is this a timing issue?

![image](https://cloud.githubusercontent.com/assets/1377228/5877141/20baaff4-a2e3-11e4-858c-133fb3463f97.png)

Note that I'm logged in as my user (an HPI admin), but the admin button doesn't appear either.

**Comment by m-brouillette** [comment]
I'm had trouble reproducing this issue locally, but I did look for potential timing issues.  What it looks to me is happening that app.user is not fetched when we're setting the the name / checking whether or not to show the admin button.
In both case we check if(app.user), which would still pass if app.user has been initialized but its fetch() call (in security.js) hasn't comeback.  When its fetch succeeds, the flag app.user.fetched is set to true.
That flag only seems good for detecting if we've had timing issue, not preventing them, which would seem to require some sort of deferred.

**Comment by mbowen000** [comment]
Can we listen to the sync event on app.user in the header module and
re-render a subview (that contains the admin button, or maybe one would
need to be created) whenever the sync is triggered?

Then we would always have the latest info from app.user..

just a thought..

**Comment by m-brouillette** [comment]
Moved the Backbone Navigates in security.js that take you to HPI into the success callback of app.user.fetch(), so that we can't get into HPI before app.user is loaded.

I think the approach @mbowen000 preferable if we have time to take the Knockout out of header.js somewhere down the line.  Since its almost all ko.observables now I used my solution, but could still see this timing issue when its fetching app.user off a cookie, which now shows an error rather than "Hello, undefined".

Leaving this open for awhile since it is hard to reproduce.
HPI REV - 11614
CR - ZK
